subsys/bluetooth/host/iso.c

```diff
@@ -700,34 +700,52 @@ static int bt_iso_setup_data_path(struct bt_conn *conn)
 {
 	int err;
 	struct bt_iso_chan *chan;
-	struct bt_iso_chan_path path = {};
+	struct bt_iso_chan_path default_path = { .pid = BT_ISO_DATA_PATH_HCI };
 	struct bt_iso_data_path out_path = {
 		.dir = BT_HCI_DATAPATH_DIR_CTLR_TO_HOST };
 	struct bt_iso_data_path in_path = {
 		.dir = BT_HCI_DATAPATH_DIR_HOST_TO_CTLR };
+	struct bt_iso_chan_io_qos *tx_qos;
+	struct bt_iso_chan_io_qos *rx_qos;
 
 	chan = SYS_SLIST_PEEK_HEAD_CONTAINER(&conn->channels, chan, node);
 	if (!chan) {
 		return -EINVAL;
 	}
 
-	in_path.path = chan->qos->tx->path ? chan->qos->tx->path : &path;
-	out_path.path = chan->qos->rx->path ? chan->qos->rx->path : &path;
+	tx_qos = chan->qos->tx;
+	rx_qos = chan->qos->rx;
 
-	if (!chan->qos->tx) {
+	in_path.path = tx_qos && tx_qos->path ? tx_qos->path : &default_path;
+	out_path.path = rx_qos && rx_qos->path ? rx_qos->path : &default_path;
+
+	if (!tx_qos) {
 		in_path.pid = BT_ISO_DATA_PATH_DISABLED;
 	}
 
-	if (!chan->qos->rx) {
+	if (!rx_qos) {
 		out_path.pid = BT_ISO_DATA_PATH_DISABLED;
 	}
 
-	err = hci_le_setup_iso_data_path(conn, &in_path);
-	if (err) {
-		return err;
-	}
+	if (conn->iso.is_bis) {
+		/* Only set one data path for BIS as per the spec */
+		if (tx_qos) {
+			return hci_le_setup_iso_data_path(conn, &in_path);
+
+		} else {
+			return hci_le_setup_iso_data_path(conn, &out_path);
+		}
+
+	} else {
+		/* Setup both directions for CIS*/
+		err = hci_le_setup_iso_data_path(conn, &in_path);
+		if (err) {
+			return err;
+		}
+
+		return hci_le_setup_iso_data_path(conn, &out_path);
```

Comment:
Is it a problem if setting up the out-path fails and the in-path succeeded, would it then be needed to remove the in-path again?

Reply:
If this function fails, we disconnect the ISO connection. We automatically remove all data paths in the disconnect callback. 

The only thing that maybe need to be improved, is that we don't check if a data path has been setup before attempting to remove it (e.g. in case this fails), so if this fails, then remove data path will also fail (we ignore errors on that at the moment though). 

I'd rather leave a comment to get that fixed in another PR, so we can fix the issues as hand first.